plugins/terraform/provider.go

```diff
@@ -1,15 +1,15 @@
 package main
 
 import (
-	"time"
-
 	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/terraform"
 	"github.com/quintilesims/layer0/cli/client"
+	"time"
 )
 
 var defaultTimeout = time.Minute * 15
 
-func Provider() *schema.Provider {
+func Provider() terraform.ResourceProvider {
```

Comment:
I'd say do whatever the builtin terraform providers do. Pending that, best practice in go when building constructors is to return the object literal, not an interface - so I'd expect `func Provider() *schema.Provider`